compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -463,29 +463,15 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             let cause = obligation.derived_cause(ObligationCauseCode::BuiltinDerived);
 
             assert_eq!(obligation.predicate.polarity(), ty::PredicatePolarity::Positive);
-            let trait_ref =
-                self.infcx.enter_forall_and_leak_universe(obligation.predicate).trait_ref;
-            let trait_obligations = self.impl_or_trait_obligations(
```

Comment:
To be very clear, this `impl_or_trait_obligations` returns a single predicate, which is this one here:


https://github.com/rust-lang/rust/blob/446649d46383734e9e047d072a482720216dd59c/compiler/rustc_hir_analysis/src/collect/predicates_of.rs#L43-L70

Arguably that predicate shouldn't even be in the `predicates_of` and perhaps should be put into the `param_env` of traits and trait items... but that seems a lot harder to do.